File: Python/Memory.py
from Hardware import Hardware
# mengimport kelas hardware


class Memory(Hardware):
    # kelas yang digunakan untuk mengimplementasikan kelas Memory dan juga
    # turunan dari kelas Hardware
    def __init__(self):
        # konstruktor
        self.frequency = ""
        self.memorySize = 0
        self.supportsCuda = ""

    def setFrequency(self, frequency):
        # mengeset nilai atribut frequency
        self.frequency = frequency

    def getFrequency(self):
        # mengembalikan nilai atribut frequency
        return self.frequency

    def setMemorySize(self, memorySize):
        # mengeset nilai atribut memorySize
        self.memorySize = memorySize

    def getMemorySize(self):
        # mengembalikan nilai atribut memorySize
        return self.memorySize

    def setSupportsCuda(self, supportsCuda):
        # mengeset nilai atribut supportsCuda
        self.supportsCuda = supportsCuda

    def getSupportsCuda(self):
        # mengembalikan nilai atribut supportsCuda
        return self.supportsCuda

    def printMemory(self):
        # menampilkan atribut memory
        print("Frequency: " + self.getFrequency())
        print("Memory Size: " + str(self.getMemorySize()))
        print("Supports Cuda: " + self.getSupportsCuda())

File: PHP/Memory.php
<?php
include("Hardware.php");
# import class Hardware

class Memory extends Hardware
# kelas yang digunakan untuk mengimplementasikan Memory dan juga turunan dari kelas Hardware
{
    # private atribute dari kelas Memory
    private $frequency = "";
    private $memorySize = 0;
    private $supportsCuda = "";

    # konstrutor
    public function __construct()
    {
    }

    # mengeset nilai atribut frequency
    public function setFrequency($frequency)
    {
        $this->frequency = $frequency;
    }

    # mengembalikan nilai atribut frequency
    public function getFrequency()
    {
        return $this->frequency;
    }

    # mengeset nilai atribut memorySize
    public function setMemorySize($memorySize)
    {
        $this->memorySize = $memorySize;
    }

    # mengembalikan nilai atribut memorySize
    public function getMemorySize()
    {
        return $this->memorySize;
    }

    # mengeset nilai atribut supportsCuda
    public function setSupportsCuda($supportsCuda)
    {
        $this->supportsCuda = $supportsCuda;
    }

    # mengembalikan nilai atribut supportsCuda
    public function getSupportsCuda()
    {
        return $this->supportsCuda;
    }

    # menampilkan atribut Memory
    public function printMemory()
    {
        echo "Supports Cuda: " . $this->getSupportsCuda() . "<br/>";
        echo "Memory Size: " . $this->getMemorySize() . "<br/>";
        echo "frequency: " . $this->getFrequency() . "<br/>";
    }
}
